### PR TITLE
feat: update squiggle

### PR DESCRIPTION
**Pull request by Srish-ty**
### Solved or Linked Issues

issue no. #

### Description and Changes

The mutation to update the squiggle by its id has been added.

### Your checklist for this pull request

🚨Please review the [guidelines for contributing](../README.md) to this repository.

- [ ] Make sure you are requesting to **pull a work/topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **development** (left side). Also you should start _your branch_ off _default branch_ and follow the branch naming guidelines.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation and mentioned the related issues (if appropriate)
- [x] I have added the appropriate labels, requested a review from a repository manager and assigned myself to this pull request.

### Post merge checklist

- [x] Follow steps from the [guidelines for contributing](../blob/development/CONTRIBUTING.md) to this repository.

### CODE REVIEW

File: server/schema/squiggle/squiggle.resolver.js [hunk 1]
```diff
@@ -33,6 +33,34 @@ module.exports = {
...
+      }
+      const _squiggle = await Squiggle.findByID(id);
+
+      if (!_squiggle) {
+        throw APIError('NOT_FOUND', null, { reason: 'The squiggle to update does not exist.',});
+      }
+      _squiggle.squiggleType = newSquiggleType;
+      _squiggle.content = newContent;
+  
+      await _squiggle.save();
```
Comment (120EE0692) [inline server/schema/squiggle/squiggle.resolver.js:57]: @Srish-ty do all the mongoose related work in squiggle datasource file.

**Review by 120EE0692 — CHANGES_REQUESTED** [review]

File: server/schema/squiggle/squiggle.mutation.js [hunk 2]
```diff
@@ -13,5 +13,16 @@ module.exports = new GraphQLObjectType({
       },
       resolve: createSquiggle,
     },
+    updateSquiggle: {
+      type: SquiggleType,
+      args: {
+        id: {
+          type: new GraphQLNonNull(GraphQLID),
```
[the remark this answers is not in the document]
Comment (Srish-ty) [reply server/schema/squiggle/squiggle.mutation.js:20]: imported GraphQLID

File: server/schema/squiggle/squiggle.mutation.js [hunk 3]
```diff
@@ -13,5 +13,16 @@ module.exports = new GraphQLObjectType({
...
+          description: "The squiggle's mongo ID",
+        },
+        newSquiggleType: { type: GraphQLString },
```
[the remark this answers is not in the document]
Comment (Srish-ty) [reply server/schema/squiggle/squiggle.mutation.js]: removed it. [commit](https://github.com/Monday-Morning/project-reclamation/pull/205/commits/5343a3e5818a1bea48d8116e8da4f13b2aecc696)

File: server/schema/squiggle/squiggle.resolver.js [hunk 4]
```diff
@@ -33,6 +33,34 @@ module.exports = {
...
   },
+
+  // added code to update squiggle here..
+  updateSquiggle: async (
+    _parent,
+    { id, newSquiggleType, newContent },
+    { session, authToken, decodedToken, API: { Squiggle } }
+  ) => {
+    try {
+      if (!UserPermission.exists(session, authToken, decodedToken, 'squiggle.write.update')) {
```
[the remark this answers is not in the document]
Comment (Srish-ty) [reply server/schema/squiggle/squiggle.resolver.js]: [commited changes](https://github.com/Monday-Morning/project-reclamation/pull/205/commits/549fe7e6c2115af970cef443a0b21d37f9f1a7d8)

File: server/schema/squiggle/squiggle.resolver.js (hunk 1 above)
Comment (Srish-ty) [reply server/schema/squiggle/squiggle.resolver.js]: done. [commit](https://github.com/Monday-Morning/project-reclamation/pull/205/commits/82bb74d89cb609214b7a3db08adf348703b34e7b)

**Comment by 120EE0692** [comment]
@Srish-ty resolve conflicts

**Comment by Srish-ty** [comment]
@120EE0692  resolved conflict

File: server/schema/squiggle/squiggle.datasources.js [hunk 5]
```diff
@@ -41,11 +41,26 @@ const create = async (squiggleType, content) => {
   }
 };
 
+const save = async (id, newContent) => {
```
Comment (Shurtu-gal) [inline server/schema/squiggle/squiggle.datasources.js:44]: This should be updateContent or update.

File: server/schema/squiggle/squiggle.resolver.js [hunk 6]
```diff
@@ -33,6 +33,32 @@ module.exports = {
...
     }
   },
+
+  updateSquiggle: async (
+    _parent,
+    { id, newContent },
+    { session, authToken, decodedToken, API: { Squiggle } }
+  ) => {
+    try {
+      if (!UserPermission.exists(session, authToken, decodedToken, 'squiggle.write.new')) {
```
Comment (Shurtu-gal) [inline server/schema/squiggle/squiggle.resolver.js:43]: According to [this](https://github.com/Monday-Morning/project-reclamation/blob/main/planning/v1/PERMISSIONS.md#squiggle-section) it should be squiggle.write.all.

File: server/schema/squiggle/squiggle.resolver.js [hunk 7]
```diff
@@ -33,6 +33,32 @@ module.exports = {
...
+        throw APIError('FORBIDDEN', null, {
+          reason: 'The user does not have the required permissions to update squiggles.',
+        });
+      }
+      let _squiggle = await Squiggle.findByID(id);
+
+      if (!_squiggle) {
+        throw APIError('NOT_FOUND', null, { reason: 'The squiggle to update does not exist.',});
+      }
+      _squiggle.content = newContent;
```
Comment (Shurtu-gal) [inline server/schema/squiggle/squiggle.resolver.js:53]: Any need of this ?

**Review by Shurtu-gal — CHANGES_REQUESTED** [review]

File: server/schema/squiggle/squiggle.resolver.js (hunk 7 above)
Comment (Shurtu-gal) [reply server/schema/squiggle/squiggle.resolver.js:53]: As _squiggle is anyhow getting assigned after this

**Pull request closed without merging**